department-of-veterans-affairs/vets-api
# [API-41018] POA Request Decision

## Summary

- Add `/decide` endpoint to power of attorney request controller.
- Update local BGSʼ `update_poa_request` to include secondary status (i.e., decision) and declined reason.
- Add tests and cassettes for failure and success cases.
 
## Related issue(s)

[API-41018](https://jira.devops.va.gov/browse/API-41018)

## Testing done


**Steps to test**
1. Connect to BGS.
1. In a rails console, find a `procId` by running:
   ```rb
   require 'bgs_service/veteran_representative_service'
   vrs = ClaimsApi::VeteranRepresentativeService.new(external_uid: 'x', external_key: 'y')
   vrs.read_all_veteran_representatives(type_code: '21-22', ptcpnt_id: '600052700')
   ```
   where `ptcpnt_id` is a participant ID containing at least one veteran representative. Feel free to use the above example if none come to mind.
   ```
   …
   "poaCode"=>"074",
   "postalCode"=>"35801",
   "procId"=>"13255", <--- This number is what we want
   "representativeFirstName"=>nil,
   "representativeLastName"=>nil,
   …
   "secondaryStatus"=>"Obsolete",
   …
   ```
   Note the `secondaryStatus`, which will be important later.
1. With a rails server running, send the following request:
    ```
    curl --location 'http://localhost:3000/services/claims/v2/veterans/power-of-attorney-requests/decide' \
    --header 'Content-Type: application/json' \
    --header 'Authorization: Bearer ******' \
    --data '{
        "data": {
            "attributes": {
                "procId": "13255",
                "decision": "accepted"
            }
        }
    }'
    ```
    where `procId` is the value from Step 2 and `decision` is either `accepted` or `declined`.
    - **Note**: `declinedReason` is optional.
1. Observe a successful response:
    ```
    {
        "VSOUserEmail": null,
        "VSOUserFirstName": "vets-api",
        "VSOUserLastName": "vets-api",
        "dateRequestActioned": "2024-10-15T23:52:26-05:00",
        "declinedReason": null,
        "procId": "13255",
        "secondaryStatus": "ACC"
    }
    ```
1. In a rails console, verify the `secondaryStatus` was updated using the same method as Step 2:
   ```rb
   vrs.read_all_veteran_representatives(type_code: '21-22', ptcpnt_id: '600052700')
   ```
   Result:
   ```
   …
   "procId"=>"13255",
   "representativeFirstName"=>nil,
   "representativeLastName"=>nil,
   …
   "secondaryStatus"=>"Accepted",
   …
   ```

## Screenshots

N/A

## What areas of the site does it impact?

- Power of attorney request controller
- Local BGS > Manage Representative Service

## Acceptance criteria


## Requested Feedback

As always, feedback on routes/functional style of programming welcome.
